torch/lib/c10d/ProcessGroupMPI.cpp

```diff
@@ -102,6 +102,25 @@ std::vector<at::Tensor> ProcessGroupMPI::WorkMPI::result() {
   }
 }
 
+c10::intrusive_ptr<c10::ivalue::Future> ProcessGroupMPI::WorkMPI::getFuture() {
+  return future_;
+}
+
+void ProcessGroupMPI::WorkMPI::finishCompleteFuture(std::exception_ptr eptr) {
+  if (eptr) {
+    future_->setError(eptr);
+    future_->markCompleted();
```

Comment:
Is this correct? IIUC, `setError` would already mark the future as completed. Then, if we call `markCompleted` again, it will throw? See the code below:

https://github.com/pytorch/pytorch/blob/dd6b9665bf3c8e447ed6680772f687d2041ebe25/aten/src/ATen/core/ivalue_inl.h#L575

https://github.com/pytorch/pytorch/blob/dd6b9665bf3c8e447ed6680772f687d2041ebe25/aten/src/ATen/core/ivalue_inl.h#L374-L377

Reply:
Curious, is there a way to test this code path?

Reply:
we'd need to inject exceptions in ProcessGroupMPI in this case.